src/features/hotspots/checklist/HotspotChecklistCarousel.tsx

```diff
@@ -12,62 +13,97 @@ export type ChecklistItem = {
+  const colors = useColors()
+  const isAndroid = Platform.OS === 'android'
 
   const keyExtractor = useCallback((item: ChecklistItem, index: number) => {
     return `${item.key}.${index}`
   }, [])
 
-  const isAndroid = Platform.OS === 'android'
+  const onScrollEnd = useCallback(
+    (e) => {
+      const windowWidth = Dimensions.get('window').width
+      const pageNum = Math.min(
+        Math.max(
+          Math.floor(e.nativeEvent.contentOffset.x / windowWidth + 0.5),
+          0,
+        ),
+        checklistData.length,
+      )
+      if (slideIndex !== pageNum) {
+        setSlideIndex(pageNum)
+      }
+    },
+    [checklistData.length, slideIndex],
+  )
+
   const renderItem = useCallback(
-    (item: { item: ChecklistItem }) => (
-      <HotspotChecklistItem
-        title={item.item.title}
-        description={item.item.description}
-        complete={item.item.complete}
-        showAuto={item.item.showAuto}
-        autoText={item.item.autoText}
-        completeText={item.item.completeText}
-        background={item.item.background}
-        isAndroid={isAndroid}
-      />
-    ),
-    [isAndroid],
+    (item: { index: number; item: ChecklistItem }) => {
+      return (
+        <HotspotChecklistItem
+          index={item.index}
+          title={item.item.title}
+          description={item.item.description}
+          complete={item.item.complete}
+          showAuto={item.item.showAuto}
+          autoText={item.item.autoText}
+          itemKey={item.item.key}
+          percentComplete={percentComplete}
+          isAndroid={isAndroid}
+          totalCount={checklistData.length}
+        />
+      )
+    },
+    [isAndroid, percentComplete, checklistData.length],
   )
-  if (isAndroid) {
-    return (
-      <FlatList
-        data={checklistData}
-        renderItem={renderItem}
-        horizontal
-        keyExtractor={keyExtractor}
-        pagingEnabled
-        showsHorizontalScrollIndicator={false}
-      />
-    )
-  }
 
   return (
-    <Carousel
-      layout="default"
-      firstItem={firstIndex === -1 ? 0 : firstIndex}
-      activeSlideAlignment="center"
-      vertical={false}
-      data={checklistData}
-      renderItem={renderItem}
-      sliderWidth={wp(100)}
-      itemWidth={wp(90)}
-      inactiveSlideScale={1}
-    />
+    <>
+      {isAndroid ? (
+        <FlatList
+          data={checklistData}
+          renderItem={renderItem}
+          horizontal
+          keyExtractor={keyExtractor}
+          pagingEnabled
+          showsHorizontalScrollIndicator={false}
+          onMomentumScrollEnd={onScrollEnd}
+        />
+      ) : (
+        <Carousel
+          layout="default"
+          firstItem={0}
+          activeSlideAlignment="center"
+          vertical={false}
+          data={checklistData}
+          renderItem={renderItem}
+          sliderWidth={wp(100)}
+          itemWidth={wp(90)}
+          inactiveSlideScale={1}
+          onScrollIndexChanged={(i) => setSlideIndex(i)}
```

Comment:
This could be `onScrollIndexChanged={setSlideIndex}`

Reply:
updated